zerver/openapi/curl_param_value_generators.py

```diff
@@ -209,3 +209,10 @@ def upload_custom_emoji() -> Dict[str, object]:
     return {
         "filename": "zerver/tests/images/animated_img.gif",
     }
+
+@openapi_param_value_generator(["/users/{user_id}:delete"])
+def deactivate_user() -> Dict[str, object]:
+    client = helpers.nonreg_user("test")
```

Comment:
I don't think this creates a user; it just gets a user that happens to have been created by `create_user` earlier in the test sequence.  I'd prefer to use `helpers.register(...)` to create a new user during this test, and then use `helpers.nonreg_user()` to get the user account.  And we should use a different user than the one used in the `create_user` test. 

As a sidenote, in your version, this local variable should be called `user_profile`, not `client`, to properly hint its type.

Reply:
@timabbott  I am getting this error when I am using `helpers.register(...)`


![Screenshot from 2020-09-15 09-44-11](https://user-images.githubusercontent.com/43513378/93164802-0c9bfc80-f738-11ea-8cc3-f6b34f3b2162.png)


**While if i user `UserProfile.objects.create(...)` test runs well.**


![test](https://user-images.githubusercontent.com/43513378/93165088-c4310e80-f738-11ea-82c2-47589eab722d.png)

Should I use this ?

Reply:
I'd recommend a code block rather than screenshots for communicating about code; it's a lot more readable.  I would not recommend using that function -- it doesn't create related models in a way that's invalid.  `do_create_user` would be a fine option.